services/api/src/routes/auth.js

```diff
@@ -74,9 +74,16 @@ router
 
       const authTokenId = generateTokenId();
       await user.updateOne({ loginAttempts: 0, authTokenId });
-      ctx.body = { data: { token: createAuthToken(user.id, authTokenId) } };
+      ctx.body = { data: { token: createAuthToken(user.id, user.authTokenId) } };
     }
   )
+  .post('/logout', authenticate({ type: 'user' }), fetchUser, async (ctx) => {
+    const user = ctx.state.authUser;
+    await user.updateOne({
+      authTokenId: generateTokenId(),
```

Comment:
sorry a bit confused though... shouldn't this do an `$unset`?

Reply:
yeah i guess it could be, what do you prefer, the way its working now it would resolve in the same thing, i almost prefer it like this because it avoid issues undefined === undefined, i do know you have check if `if(jwt.jti && ...)`
let me know

Reply:
No an `authTokenId` can easily be undefined, for example when the user is first created but hasn't logged in yet. It's very expected. This is already set up to work [here](https://github.com/bedrockio/bedrock-core/blob/master/services/api/src/utils/middleware/authenticate.js#L62) and there should be tests on it. Meanwhile if we set a new tokenId then it means that in theory some mysterious JWT token that no one has access to could log them in. Even if that's not possible in practice it makes the intent very unclear.

Reply:
done